### PR TITLE
Cross compile support for cpptools integration.

### PR DESCRIPTION
**Pull request by bjosa**
# Cross Compile support for cpptools integration

Added compile flags and sysroot to the compiler path for cpptools.

## This change addresses [#637]

Generally all cross compiling toolchains that depend on special compiler flags and a sysroot. These flags have to be published to the cpptools compilePath option. The cpptools extension then parses the compiler with These flags for the right include paths and defines.

### This changes cpptools integration

The following changes are proposed:

- Add compile flags to cpptools compilePath
- Add sysroot to cpptools compilePath

### CODE REVIEW

**Comment by bjosa** [comment]
Maybe related to #604, #599

File: src/cpptools.ts [hunk 1]
```diff
@@ -208,17 +208,28 @@ export class CppConfigurationProvider implements cpt.CustomConfigurationProvider
         newBrowsePath.push(includePathItem);
       }
     }
+
+    let cpptoolsCompilerPath = (flags.length === 0) ? comp_path : `${comp_path} ${flags.join(' ')}`;
+    if (sysroot && cpptoolsCompilerPath.indexOf("--sysroot") < 0) {
+      if (sysroot.match(/\s/)) {
+        cpptoolsCompilerPath += ` "--sysroot=${sysroot}"`;
```
Comment (bobbrow) [inline src/cpptools.ts]: The opening `"` seems to be in the wrong place.  Shouldn't it be after `--sysroot=`?  And is `sysroot` guaranteed not to have additional `"`'s in it?

Comment (bjosa) [reply src/cpptools.ts]: Thank's for looking over this. Think this should be ok. The `sysroot` option points to an path. The path will be quoted if it contains a space. The double quote surrounds `--sysroot` so the whole term, option and path, is interpreted as an command line argument.

**Comment by icegood** [comment]
I don't get how to use it. Please, update official documentation. At least I don't see any way to set sysroot somewhere in options.

**Comment by bjosa** [comment]
Just provide a `cmake-kits.json` file with the specific settings for your toolchain. See the [cmake-kits](https://github.com/microsoft/vscode-cmake-tools/blob/develop/docs/kits.md) section of the documentation for more information.

Example cmake-kits entry with an CMake toolchain file. The CMake toolchain file of the sdk then defines the sysroot path and all neccessary compiler flags:

```json
{
    "name": "arm-linux-gnueabihf",
    "toolchainFile": "/home/user/sdk/toolchain.cmake",
    "cmakeSettings": {
        "CMAKE_EXPORT_COMPILE_COMMANDS": "ON",
        "CMAKE_STAGING_PREFIX": "${workspaceFolder}/build/${buildKit}/out",
        "CMAKE_PREFIX_PATH": [
            "${workspaceFolder}/build/${buildKit}/out"
        ]
    },
    "preferredGenerator": {
        "name": "Ninja"
    }
}
```

**Pull request merged**